### PR TITLE
[integer-overflow] enforce some checked operations

### PR DESCRIPTION
**Pull request by mimoo**

### CODE REVIEW

File: common/lcs/src/de.rs [hunk 1]
```diff
@@ -513,21 +514,24 @@ impl<'de, 'a> de::MapAccess<'de> for MapDeserializer<'a, 'de> {
...
-            Ok(None)
-        } else {
-            let previous_input_slice = &self.de.input[..];
-            let key_value = seed.deserialize(&mut *self.de)?;
-            let key_len = previous_input_slice.len() - self.de.input.len();
-            let key_bytes = &previous_input_slice[..key_len];
-            if let Some(previous_key_bytes) = self.previous_key_bytes {
-                if previous_key_bytes >= key_bytes {
-                    return Err(Error::NonCanonicalMap);
+        match self.remaining.checked_sub(1) {
```
Comment (matbd) [inline common/lcs/src/de.rs:517]: lgtm

File: network/memsocket/src/lib.rs [hunk 2]
```diff
@@ -105,17 +105,17 @@ impl MemoryListener {
...
-                if switchboard.1 == std::u16::MAX {
-                    switchboard.1 = 1;
-                } else {
-                    switchboard.1 += 1;
-                }
+
+                switchboard.1 = match switchboard.1.checked_add(1) {
+                    None => 1,
+                    Some(i) => i,
+                };
```
Comment (davidiw) [inline network/memsocket/src/lib.rs:118]: functors....

unwrap_or_else(1)

**Review by davidiw — APPROVED** [review]

**Pull request closed without merging**